### PR TITLE
feat(header): L3-4484 added close menu functionality

### PR DESCRIPTION
**Pull request by constantinehuzenko**
**Jira ticket**

[L3-4484](https://phillipsauctions.atlassian.net/browse/L3-4484)

**Screenshots**

https://github.com/user-attachments/assets/ef51d9c3-27d1-421e-a0a4-734c7aea41e4

**Summary**

Closing `Navigation` on `NavigationItem` click

**Change List (describe the changes made to the files)**

- src/components/Navigation/NavigationItem/NavigationItem.tsx: added closing functionality
- src/components/Navigation/NavigationList/NavigationList.tsx: passed closing callback
- src/site-furniture/Header/Header.tsx: moved some functions to hooks to increase test coverage
- src/patterns/LanguageSelector/LanguageSelector.tsx: moved function out to increase test coverage

**Acceptance Test (how to verify the PR)**

- link seldon to remix project and verify that mobile and desktop header collapses when you switch between pages 

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-4484]: https://phillipsauctions.atlassian.net/browse/L3-4484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by adietrich3074 — APPROVED** [review]
LGTM but is the Language Selector stuff supposed to be a part of this?

File: src/components/Navigation/NavigationItem/NavigationItem.tsx [hunk 1]
```diff
@@ -55,15 +61,22 @@ const NavigationItem = forwardRef<HTMLLIElement, NavigationItemProps>(
...
+    const { closeMenu } = React.useContext(HeaderContext);
+
     return (
       <li
         {...props}
-        onClick={onClick}
+        onClick={(e) => {
+          closeMenu?.();
+          closeSubmenu?.();
+          onClick?.(e);
```
Comment (davidicus) [inline src/components/Navigation/NavigationItem/NavigationItem.tsx:78]: If we just need to pass in the state update function, is it possible to leverage the closeMenu or onClick handlers?

Comment (davidicus) [reply src/components/Navigation/NavigationItem/NavigationItem.tsx:78]: Can't we just pass in whatever callback we want, one that closes all menus. So if they are all state setters
```js
onCloseMenu = () => {
  setIsMobileMenuOpen(false);
  setIsDropdownExpanded(false);
 }
```
then in the component we can just call `onCloseMenu` on click?

Comment (davidicus) [reply src/components/Navigation/NavigationItem/NavigationItem.tsx:78]: Just wondering if we need to increase our Prop API or if we can just leverage the onClick that is already applied.

File: src/components/Navigation/NavigationItem/NavigationItem.tsx [hunk 2]
```diff
@@ -60,10 +62,15 @@ const NavigationItem = forwardRef<HTMLLIElement, NavigationItemProps>(
     },
     ref,
   ) => {
+    const { closeMenu } = React.useContext(HeaderContext);
```
Comment (davidicus) [inline src/components/Navigation/NavigationItem/NavigationItem.tsx:65]: Can this be moved to the src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx so all the close logic is being added to the onClick

**Comment by davidicus** [comment]
This is working good but I am seeing that in mobile menu when you click a link it closes the menu but not the accordion

**Comment by constantinehuzenko** [comment]
> This is working good but I am seeing that in mobile menu when you click a link it closes the menu but not the accordion @constantinehuzenko was that supposed to be part of this fix?

no I didn't think about this test case, will fix

File: src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx [hunk 3]
```diff
@@ -16,8 +17,10 @@ export interface NavigationItemTriggerProps extends ComponentProps<'li'> {
 }
 
 const MobileNavigationItemTrigger = ({ id, label, children }: NavigationItemTriggerProps) => {
+  const { isMenuOpen } = React.useContext(HeaderContext);
```
Comment (davidicus) [inline src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx:20]: Nitpick: we could have pulled this from the context we are using the the component and passed to the mobile trigger.

**Review by davidicus — APPROVED** [review]

**Pull request merged**